bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/AnonymousUserSecurityContext.java

```diff
@@ -10,7 +10,7 @@
  *
  * SPDX-License-Identifier: EPL-2.0
  */
-package org.openhab.core.io.rest.auth.internal;
+package org.openhab.core.io.rest.auth;
```

Comment:
This security context was introduced in OH 3.0 and it does work around other security related API/SPI which was present earlier - `AuthenticationSecurityContext`. If you decide to go with jaxrs API as a main driver for security in websockets in servlets you shall deprecate other classes.

Reply:
I do not understand this comment, nether which are the classes that should be deprecated :(
Could you elaborate it?
Or we can let it for the end if it's not important.

Reply:
Sorry, I did not intend to confuse you. There are old parts in code which use `AuthenticationManager`, we used to have a "secured servlets" which didn't get adopted in 2.5, so they are lingering. However they still keep a reference on how to interact with auth manager:
https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.io.http.auth/src/main/java/org/openhab/core/io/http/auth/internal/AuthenticationHandler.java#L80L99

See line 86 in above example, it gives you back `Authentication` which is OH specific representation of authenticated user/token etc. Going with this path you are free of jaxrs or http specifics - you provide credentials to authentication manager and authentication manager will seek for `AuthenticationProvider` which will be capable of using it. Since there might be multiple providers for same credentials manager will fail only if all of them fail to authenticate user.

Its a bit more complex than filter, but it was designed way earlier for variety of scenarios.

My point is - if there is no further use of `AuthenticationManager` and providers, and instead we go with `AuthFilter` we shall deprecate earlier and surrounding infrastructure (Authentication, AuthenticationProvider). If intention is to keep above then we shall abstain from exposing internal implementation details of jaxrs to other places.

Reply:
Thank you for the explanation, now it all makes sense.